### PR TITLE
Migrate tools from iDynTree

### PR DESCRIPTION
**Pull request by traversaro**
This migrates all YARP-depending tool from iDynTree, making iDynTree effectively YARP independent except for a few tests that are not relevant for actual production use. As this code is effectively just importing existing doe, I don't think it is a great idea to modify too much of it, so even if some nice-to-have changes emerges during the review, I would prefer to minimize the changes done in this PR, and have subsequent PR to fix any problem may emerge.

Needs to be rebased on top of https://github.com/robotology/idyntree-yarp-tools/pull/11 once that is merged.

### CODE REVIEW

**Comment by traversaro** [comment]
> Needs to be rebased on top of #11 once that is merged.

This was done, the PR is now ready for review.

File: CMakeLists.txt [hunk 1]
```diff
@@ -60,10 +60,21 @@ set(CMAKE_CXX_EXTENSIONS OFF)
...
+option(IDYNTREE_YARP_TOOLS_USES_ICUB_MAIN "Compile the tools that depend on icub-main." ON)
+option(IDYNTREE_YARP_TOOLS_USES_QT "Compile the tools that depend on Qt." ON)
+
+if(IDYNTREE_YARP_TOOLS_USES_ICUB_MAIN)
+  find_package(ICUB REQUIRED)
+endif()
+
+if(IDYNTREE_YARP_TOOLS_USES_QT)
+  find_package(Qt5 COMPONENTS Widgets Charts REQUIRED)
+endif()
```
Comment (S-Dafarra) [inline CMakeLists.txt]: Would it be possible to have some basic mechanism to check if those dependencies are actually available before setting the default value of the option? Then, if the user sets it manually, it is totally ok to crash and burn

Comment (traversaro) [reply CMakeLists.txt]: Yes, I wanted to avoid that due to the additonal complexity in CMake, but it is trivial to do so.

Comment (traversaro) [reply CMakeLists.txt]: Done: https://github.com/robotology/idyntree-yarp-tools/pull/12/commits/6d66d4e05ce43e33ef85d644d881ec4c5f661aaf .

**Review by S-Dafarra — APPROVED** [review]

**Pull request merged**